### DISCUSSION TITLE
Report an issue

### DISCUSSION THREAD

**Discussion by otseng** [eliranwong/UniqueBible#133] [General]
If you find a defect in UBA, please [create an issue ticket](https://github.com/eliranwong/UniqueBible/issues).

**Comment by Elvis75Cristiano** [comment]
HOLA MUY BUENOS QUE DIOS LES BENDIGA MUCHO. DECIRLES QUE DISFRUTO MUCHO LA APLICACION UNIQUEBIBLE LES AGRADESCO MUCHO POR SU TRABAJO. RESULTA QUE AYER INTENTE CAMBIAR EL LENGUAGE DEL PROGRAMA DE INGLES A ESPAÑOL Y NO ME QUISO INICIAR MAS. ME PONE ESTE MENSAJE: 
PS C:\Users\PC> cd C:\Users\PC\UniqueBible
PS C:\Users\PC\UniqueBible> python uba.py
PS C:\Users\PC\UniqueBible> Traceback (most recent call last):
  File "C:\Users\PC\UniqueBible\main.py", line 378, in <module>
    config.mainWindow = MainWindow()
  File "C:\Users\PC\UniqueBible\gui\MainWindow.py", line 118, in __init__
    self.setupMenuLayout(config.menuLayout)
  File "C:\Users\PC\UniqueBible\gui\MainWindow.py", line 216, in setupMenuLayout
    getattr(windowClass, 'create_menu')(self)
  File "C:\Users\PC\UniqueBible\gui\FocusMainWindow.py", line 42, in create_menu
    addMenuItem(subMenu, feature, self, action)
  File "C:\Users\PC\UniqueBible\gui\MenuItems.py", line 25, in addMenuItem
    return menu.addAction(QAction(config.thisTranslation[feature] if translation else feature, object, triggered=action, shortcut=shortcut))
KeyError: 'bibleCollections'
 
AYUDA POR FAVOR¡¡¡ DIOS LES BENDIGA.

**Comment by onegladiator4u** [comment]
Is google drive backup working ? or it's just for me?

**Reply by otseng** [reply]
I don't think google drive support was added.

**Reply by onegladiator4u** [reply]
there was option to backup note to google drive using plugin. Last week onward it's not uploading

**Reply by otseng** [reply]
Oh, I've never used that before.  Are you getting any error messages?

**Reply by onegladiator4u** [reply]
It says failed to upload bible notes.
i checked uba wiki to download json file. But it totally looks different to create drive API. I managed to download and replace the file again. it didn't work too. Don't know whether i downloaded correctly

**Reply by otseng** [reply]
I was able to get it working for me, but the Google process setup has changed from the wiki. 

Steps:
1. Goto [Google cloud console](https://console.cloud.google.com) and select a project if one already created.  Otherwise create a project. 
2. Enable "Google Drive API"
3. Create Oauth credentials and download file and rename it to credentials.json
4. Link Google Drive API to your OAuth credentials (it seemed to auto link them for me)
5. In Oauth consent screen, toggle Publishing status to "In Production"

It was the last step for me that was not obvious.  Here's a screenshot of it:
![Screenshot from 2022-07-13 09-58-56](https://user-images.githubusercontent.com/806056/178754406-452af8e3-93b2-4fc4-89c4-cdbaa5fc4300.png)

**Reply by eliranwong** [reply]
@otseng can you help revising the wiki page?  Many thanks

**Reply by otseng** [reply]
Added:
https://github.com/eliranwong/UniqueBible/wiki/Setting-up-Google-Drive-API

Also updated:
https://github.com/eliranwong/UniqueBible/wiki/Notes-Backup-with-Google-Drive

**Reply by onegladiator4u** [reply]
When trying to create OAuth client id its showing this
![image](https://user-images.githubusercontent.com/48617390/179402167-f0919f42-e5de-4a98-9010-c9d3ccccfd9c.png)

Config consent screen looks like this and i could only select External , 
![image](https://user-images.githubusercontent.com/48617390/179402328-767d34f4-1637-4544-ac43-c8d0c4bccc46.png)
![mXMtvnM6Cm](https://user-images.githubusercontent.com/48617390/179402761-3ea7a8c5-2b7b-4310-84cd-c92c36e48c06.png)
What to do in this ?

**Reply by otseng** [reply]
Set user type to external.  
Enter "Unique Bible App" for name.
Enter your email for support email.
Leave everything else blank.

**Reply by onegladiator4u** [reply]
i did like it . I didn't get google sign in window or anything.
Just the same message came
![image](https://user-images.githubusercontent.com/48617390/179452642-3c3cbde3-5f88-4c60-a141-a74e66a038b9.png)

**Reply by otseng** [reply]
I just added some debugging code to it.  Can you update and rerun the backup and see what the error message says in the console?

**Reply by onegladiator4u** [reply]
![image](https://user-images.githubusercontent.com/48617390/179568197-a9956955-ea26-4c31-854f-654edb657e4b.png)

**Reply by otseng** [reply]
Try stopping UBA, delete the file "token.pickle", then restart UBA.

**Reply by onegladiator4u** [reply]
It opened google for authentication. After that it showed

![firefox_Mag1akij75](https://user-images.githubusercontent.com/48617390/179673628-4299261f-fd7c-495f-97ef-e7f809910788.png)
 But same message came that it failed .
console showed
![image](https://user-images.githubusercontent.com/48617390/179674246-e2d79faf-e238-44a6-9805-21f92e53e6b2.png)

**Reply by otseng** [reply]
After loading your venv, run `pip install --upgrade six`.
See this for more info on this error:
https://stackoverflow.com/questions/63902467/module-six-moves-has-no-attribute-collections-abc

**Reply by onegladiator4u** [reply]
is this how to load venv?
python -m venv venv

.\venv\Scripts\activate

**Reply by onegladiator4u** [reply]
i did like it, this came
![image](https://user-images.githubusercontent.com/48617390/179800396-1420b118-37ac-4d04-953a-9bd148f069c7.png)

and same error occurred. i might be doing something wrong with venv

just so you know i have another venv folder named `venv_Windows_3.9.1`

**Reply by otseng** [reply]
Yes, use the venv_Windows_3.9.1 not the venv.  So run "./venv_Windows_3.9.1/Script/activate".   You don't need the "venv" folder so that can be deleted.

**Reply by onegladiator4u** [reply]
upgraded six
![image](https://user-images.githubusercontent.com/48617390/179820080-d905aaff-0d04-4995-85bf-94f40ce4ebb7.png)
still error , looks same 
![image](https://user-images.githubusercontent.com/48617390/179821638-ad397e15-5597-43be-851a-182ff0fd57f3.png)

**Reply by onegladiator4u** [reply]
bro, did you notice that uba is using location `C:\Python39\Lib\site-packages\googleapiclient` but six was updated on `F:` drive `/venv_Windows_3.9.1` ?

**Reply by otseng** [reply]
That is strange.  It should all run from code in F: drive.

Can you run `deactivate` then run `pip uninstall google-api-python-client` to remove it from C:

Then activate the venv_Windows_3.9.1 and run `pip install --upgrade google-api-python-client`.

**Reply by onegladiator4u** [reply]
can you explain the process to deactivate it? it would be great, like where should I run deactivate and all.

**Reply by otseng** [reply]
When you first open a console (command shell), it will use the Python located in C: drive (deactivated mode).

When you run `./venv_Windows_3.9.1/Script/activate` it will use Python located in F: drive (where your UBA is located).  This will be in activated mode.

When in activated mode, you can switch back to the deactivated mode by running `deactivate`.

The bottom line is all your libraries should be updated in F: drive, not C: drive.

**Reply by onegladiator4u** [reply]
Thank you so brother it worked . 
Brother i had problem with qtmaterials just like this. It was using C: drive. site packages . Is there a permanent solution to use from F:. this is a temporary solution right?

**Reply by otseng** [reply]
Great.  I'm hoping it would be a permanent solution.

**Reply by onegladiator4u** [reply]
Hi @otseng ,
i recently reinstalled windows. Now when I try to upload to google drive it shows like this.
![python_M2M2YpZ3om](https://user-images.githubusercontent.com/48617390/198255715-9a82e38c-5e46-4bcc-b417-529870ab3666.png)

PS G:\BIBLE STUDY\App\UniqueBible> pip tool updated!
Installing missing module 'chinese-english-lookup' ...
Optional feature 'Chinese-to-English word definition' is not enabled.
Run 'pip3 install chinese-english-lookup' to install it first.
INFO: Could not find files for the given pattern(s).
Installing missing module 'pickley' ...
INFO: Could not find files for the given pattern(s).
Optional feature 'Automate installation of standalone python CLIs' is not enabled.
Run 'pip3 install pickley' to install it first.
INFO: Could not find files for the given pattern(s).
b"'C:\\Program' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"
b"'C:\\Program' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"
b"'C:\\Program' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"
b"'C:\\Program' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"
b"'C:\\Program' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"
b"'C:\\Program' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"

Can you help ?

**Reply by otseng** [reply]
Can you replace the file `plugins/menu/Google Drive Backup.py` with below and rerun and see what it says?

[Google Drive Backup.zip](https://github.com/eliranwong/UniqueBible/files/9895002/Google.Drive.Backup.zip)

**Reply by onegladiator4u** [reply]
I just had to reinstall windows again. So i'll let you know after i set up everything again . Thanks bro

**Reply by onegladiator4u** [reply]
Google authentication page opened but it doesn't go after that
![firefox_pMPymgprvh](https://user-images.githubusercontent.com/48617390/200637593-e97cb9b1-e373-476f-802a-05731ff04e0a.png)
UBA is not responsive 
in powershell its like this 

PS D:\App\UniqueBible> pip tool updated!
Installing missing module 'PySide2' ...
Installed!queBible\venv_Windows_3.11.0\Lib\site-packages\qtpy\__init__.py:267: RuntimeWarning:
D:\App\UniqueBible\venv_Windows_3.11.0\Lib\site-packages\qtpy\__init__.py:267: RuntimeWarning: Selected binding "pyside2" could not be found, using "pyqt5"
  warnings.warn('Selected binding "{}" could not be found, '
Installing missing module 'html-text' ...enabled.
Optional feature 'Read html text' is not enabled.
Run 'pip3 install html-text' to install it first.
Installing missing module 'htmldocx' ...is not enabled.
Optional feature 'Convert HTML to DOCX' is not enabled.
Run 'pip3 install htmldocx' to install it first.
Installing missing module 'python-docx' ...enabled.
Optional feature 'Handle DOCX file' is not enabled.
Run 'pip3 install python-docx' to install it first.
Installing missing module 'telnetlib3' ... library' is not enabled.
Optional feature 'Telnet Client and Server library' is not enabled.
Run 'pip3 install telnetlib3' to install it first.
Installing missing module 'translate' ...t enabled.
Optional feature 'Google Translate' is not enabled.
Run 'pip3 install translate' to install it first.
Installing missing module 'yt-dlp' ...is not enabled.
Optional feature 'YouTube Downloader' is not enabled.
Run 'pip3 install yt-dlp' to install it first.
Installing missing module 'lemmagen3' ...led.
Optional feature 'Lemmatizer' is not enabled.rst.
Run 'pip3 install lemmagen3' to install it first.' ...
Installing missing module 'chinese-english-lookup' ...is not enabled.
Optional feature 'Chinese-to-English word definition' is not enabled.
Run 'pip3 install chinese-english-lookup' to install it first.
Installing missing module 'textract' ...ment' is not enabled.
Optional feature 'Extract text from document' is not enabled.
Run 'pip3 install textract' to install it first.(s).
INFO: Could not find files for the given pattern(s).
Backing up marvelData/note.sqlite

**Comment by DanWell528** [comment]
I've been waiting for the SBL Greek Testament to download for over two hours now. I know it says it might take a while, but is this expected?
Mac Version 34.11

**Reply by otseng** [reply]
As far as I know, there is no SBL Greek NT to download from UBA.  Can you show the screenshot of how you're trying to download this?

**Reply by eliranwong** [reply]
> I've been waiting for the SBL Greek Testament to download for over two hours now. I know it says it might take a while, but is this expected?
> Mac Version 34.11

@DanWell528 for your convenience, I zip two SBLGNT files that I use for you.  You need to unzip them and placed in "marvelData/bibles" folder.

[SBLGNT.bible.zip](https://github.com/eliranwong/UniqueBible/files/10967206/SBLGNT.bible.zip)

[SBLGNTl.bible.zip](https://github.com/eliranwong/UniqueBible/files/10967207/SBLGNTl.bible.zip)

**Reply by otseng** [reply]
Ah, yes, I was wrong. SBLGNT is avail.  It's a 106 MB file, so it can take awhile, but it shouldn't take 2 hours to download.

**Reply by DanWell528** [reply]
Thank you both so much!

**Comment by eliranwong** [comment]
@otseng SBLGNT is downloadable
